profiles/permissions.py

```diff
@@ -13,6 +13,17 @@ def has_object_permission(self, request, view, obj):
         return obj.person == request.user
 
 
+class RequestIsCreate(BasePermission):
+    def has_permission(self, request, view):
+        return request.method == "POST"
+
+
+class IsOwner(BasePermission):
+    def has_permission(self, request, view):
+        profile = Profile.objects.get(id=request.data.get("company"))
```

Comment:
I strongly recommend to avoid any manipulations with payload (stuff which is meant to be handled by _serializers_) in permission classes.

Current user, view kwargs (i.e. parts of URL) — but not POST/PUT/PATCH payload and not GET-params of URL.

Reply:
Fixed it, thanks.